zarr/core.py

```diff
@@ -1176,3 +1176,52 @@ def view(self, shape=None, chunks=None, dtype=None,
             a._filters = filters
 
         return a
+
+    def astype(self, dtype):
+        """Does on the fly type conversion of the underlying data.
+
+        Parameters
+        ----------
+        dtype : string or dtype
+            NumPy dtype.
+
```

Comment:
So one could technically write to disk when using `astype`. I don't know if we should add a note or warning about this. Maybe something similar to what `view` has. Please let me know if you have any thoughts on this.